shotover-proxy/src/message/mod.rs

```diff
@@ -20,325 +17,233 @@ use itertools::Itertools;
+            MessageInner::Parsed { frame, .. } => Some(frame),
+            MessageInner::Modified { frame } => Some(frame),
+        }
     }
 
-    pub fn new_no_original(details: MessageDetails, modified: bool) -> Self {
-        Message {
-            details,
-            modified,
-            original: Frame::None,
-            return_to_sender: false,
-            meta_timestamp: None,
+    pub fn into_encodable(self) -> Encodable {
+        match self.inner.unwrap() {
+            MessageInner::RawBytes { bytes, .. } => Encodable::Bytes(bytes),
+            MessageInner::Parsed { bytes, .. } => Encodable::Bytes(bytes),
+            MessageInner::Modified { frame } => Encodable::Frame(frame),
         }
     }
 
-    pub fn namespace(&self) -> Option<Vec<String>> {
-        match &self.original {
+    /// Returns None when fails to parse the message
+    pub fn namespace(&mut self) -> Option<Vec<String>> {
+        match self.frame()? {
             Frame::Cassandra(cassandra) => Some(cassandra.namespace()),
             Frame::Redis(_) => unimplemented!(),
             Frame::None => Some(vec![]),
         }
     }
 
     pub fn invalidate_cache(&mut self) {
-        // TODO: invalidate cache when we have a cache to invalidate
-    }
-
-    #[must_use]
-    pub fn to_filtered_reply(&self) -> Self {
-        Message {
-            details: MessageDetails::Unknown,
-            modified: true,
-            original: match &self.original {
-                Frame::Redis(_) => Frame::Redis(RedisFrame::Error(
-                    "ERR Message was filtered out by shotover".into(),
-                )),
-                Frame::Cassandra(frame) => Frame::Cassandra(CassandraFrame {
-                    version: frame.version,
-                    stream_id: frame.stream_id,
-                    operation: CassandraOperation::Error(ErrorBody {
-                        error_code: 0,
-                        message: "Message was filtered out by shotover".into(),
-                        additional_info: AdditionalErrorInfo::Server,
-                    }),
-                    tracing_id: None,
-                    warnings: vec![],
+        // TODO: clear message details cache fields if we ever add any
+
+        self.inner = self.inner.take().map(|x| x.invalidate_cache());
+    }
+
+    // TODO: this could be optimized to avoid parsing the cassandra sql
+    pub fn to_filtered_reply(&mut self) -> Message {
+        // TODO: set_filtered_reply
+        Message::from_frame(match self.frame().unwrap() {
+            Frame::Redis(_) => Frame::Redis(RedisFrame::Error(
+                "ERR Message was filtered out by shotover".into(),
+            )),
+            Frame::Cassandra(frame) => Frame::Cassandra(CassandraFrame {
+                version: frame.version,
+                stream_id: frame.stream_id,
+                operation: CassandraOperation::Error(ErrorBody {
+                    error_code: 0,
```

Comment:
Well that value was picked at random.
Looking through the options, maybe:
    0x2100    Unauthorized: The logged user doesn't have the right to perform
              the query.
or:
    0x2300    Config_error: The query is invalid because of some configuration issue
    
Could be better?
But I dont think it matters a whole lot.
For now I'll just set to 0x0